src/main/java/org/mskcc/cbio/oncokb/service/UserService.java

```diff
@@ -619,24 +626,36 @@ private void expireUserAccount(UserDTO userDTO){
 
     /**
      * Find the company entity associated with the user by the domain part of their
-     * email. Modifications are needed for mixed usages and micro licenses.
+     * email. If an empty Optional is returned, then there is either no company
+     * associated with the domain or the user needs to be manually approved.
+     * Otherwise, a Optional with a company is returned.
      * @param userDTO the user
      */
-    private Optional<Company> findAssociatedCompany(UserDTO userDTO) {
-        
-        if(userDTO.getCompany() != null){
-            return companyRepository.findById(userDTO.getCompany().getId());
-        }
-
-        String domain = StringUtil.getEmailDomain(userDTO.getEmail());
-        List<CompanyDomain> companyDomains = companyDomainRepository.findByName(domain);
-        if(companyDomains.size() > 0) {
-            if(companyDomains.size() == 1){
-                return companyRepository.findById(companyDomains.get(0).getCompany().getId());
-            }else{  // Mixed usage, same company domain is used 
-                // Todo
+    public Optional<Company> findAssociatedCompany(UserDTO userDTO) {
+
+        // Using the email domain, find a company, if possible.
+        String emailDomain = StringUtil.getEmailDomain(userDTO.getEmail());
+        Optional<CompanyDomain> optionalCompanyDomain = companyDomainRepository.findOneByNameIgnoreCase(emailDomain);
+        if(optionalCompanyDomain.isPresent()){
+            Set<Company> companies = optionalCompanyDomain.get().getCompanies();
+            if(companies.size() == 1){
+                Company company = companies.iterator().next();
+                // When there is only a micro tiered company, then we return empty optional.
+                if(company.getLicenseModel().equals(LicenseModel.REGULAR)){
+                    // If only a regular license model exists for domain, then auto approve user with company.
+                    return Optional.of(company);
+                }
+            }else if(companies.size() > 1){
+                // If there are mutliple companies with the domain, then we find a company with a regular license.
+                return companies
```

Comment:
This is a loophole. We should not have any domain be associated with multiple regular licensed companies. Though, this is really challenge to prevent in implementation. Let's discuss.

Reply:
I just made the adjustments to show the warning messages when domain is associated with multiple regular licensed companies. 
The domain buttons will be highlight with $warning and there will be an react-bootstrap Alert with a warning message.
Here is a screenshot:
![image](https://user-images.githubusercontent.com/59149377/144122269-28f43c0e-4a15-4954-b9e3-f93e5547c0a2.png)